packages/neuron-ui/src/utils/validators/sudtAddress.ts

```diff
@@ -18,10 +18,17 @@ export const validateSUDTAddress = ({
   if (address) {
     validateAddress(address, isMainnet)
     const parsed = ckbCore.utils.parseAddress(address, 'hex')
-    // only anyone can pay address is validated
-    if (!parsed.startsWith(LONG_TYPE_PREFIX)) {
+
+    /**
+     * Only anyone can pay address(type id ver/data ver) is under validation
+     * - data version for mainnet
+     * - type version for testnet
+     */
+    const addrType = isMainnet ? LONG_DATA_PREFIX : LONG_TYPE_PREFIX
```

Comment:
The network type doesn't necessarily determine the address type for ACP. I think we can remove this check.

Reply:
Can neuron-wallet pick the right code hash by the address type?

Reply:
It should be able to.

Reply:
Then [this method](https://github.com/nervosnetwork/neuron/blob/develop/packages/neuron-wallet/src/controllers/api.ts#L390-L392) should be refactored to provides those two kinds of code hash that UI can impose validation on the addresses.

Reply:
And [this part](https://github.com/nervosnetwork/neuron/blob/8954b7b59c4d7ed67a7d669727f246a0e1f3604a/packages/neuron-wallet/src/services/tx/transaction-generator.ts#L696) of generating ACP tx should be refactored two

Reply:
Ok, let's keep this check for now. We will come back to review these reactors are necessary if this check is removed later.